### PR TITLE
D3ASIM-3556: Export market structures to d3a-interface

### PR DESCRIPTION
**Pull request by Hassan754**
I tried my best to keep the minimum changes, I left room for styling improvements that were besides my changes and did not touch them

It's best if you can review commit by commit alongside the [d3a-interface side ](https://github.com/gridsingularity/d3a-interface/pull/253)

### CODE REVIEW

File: src/d3a/models/market/one_sided.py [hunk 1]
```diff
@@ -62,17 +62,17 @@ def __repr__(self):  # pragma: no cover
...
         when placing an offer to a market.
         :param offer_price: Price of the offer coming from the source market, in cents
-        :param original_offer_price: Price of the original offer from the device
+        :param original_price: Price of the original offer from the device
         :param energy: Energy of the offer
         :return: Updated price for the forwarded offer on this market, in cents
         """
         return self.fee_class.update_incoming_offer_with_fee(
-            offer_price / energy, original_offer_price / energy
+            offer_price / energy, original_price / energy
```
Comment (spyrostz) [inline src/d3a/models/market/one_sided.py:75]: How about `offer_price` -> `price` ?

File: src/d3a/models/market/two_sided.py [hunk 2]
```diff
@@ -67,32 +67,32 @@ def __repr__(self):  # pragma: no cover
                     self.accumulated_trade_price
                     )
 
-    def _update_new_bid_price_with_fee(self, bid_price, original_bid_price):
-        return self.fee_class.update_incoming_bid_with_fee(bid_price, original_bid_price)
+    def _update_new_bid_price_with_fee(self, bid_price, original_price):
```
Comment (spyrostz) [inline src/d3a/models/market/two_sided.py:70]: Same as above, `bid_price` -> `price`

**Review by spyrostz — DISMISSED** [review]
2 comments, LGTM

File: tests/strategies/external/utils.py [hunk 3]
```diff
@@ -71,7 +71,7 @@ def assert_bid_offer_aggregator_commands_return_value(return_value, is_offer):
     assert return_value[command_name][
         "seller" if is_offer else "buyer"] == "forecast_pv"
     assert return_value[command_name][
-        "original_offer_price" if is_offer else "original_bid_price"] == 200.0
+        "original_price" if is_offer else "original_price"] == 200.0
```
Comment (spyrostz) [inline tests/strategies/external/utils.py:74]: Minor, but I think that `return_value[command_name]["original_price"]` should suffice

**Review by spyrostz — APPROVED** [review]
LGTM, one minor comment

**Review by faizan2590 — APPROVED** [review]
LGTM

File: tests/strategies/external/utils.py (hunk 3 above)
Comment (Hassan754) [reply tests/strategies/external/utils.py]: @spyros thanks for spotting this, I have overlooked it, done!

File: src/d3a/d3a_core/export.py [hunk 4]
```diff
@@ -22,29 +22,29 @@
 import os
 import pathlib
 import shutil
+import plotly.graph_objs as go
```
Comment (spyrostz) [inline src/d3a/d3a_core/export.py:25]: Very minor, but you can organize the imports here if I see correctly

**Review by spyrostz — APPROVED** [review]
LGTM

**Pull request merged**